File: GameStudies/CPP/Menu.cpp
#include "Menu.hpp"
#include <iostream>

void Menu::adicionarOpcao(std::string opcao) {
    opcoes.push_back(opcao);
}

void Menu::mostrar() {
    for (int i = 0; i < opcoes.size(); i++) {
        std::cout << i << " - " << opcoes[i] << "\n";
    }
}

File: GameStudies/CS/Menu.cs
using System;
using System.Collections.Generic;

public class Menu
{
    private List<string> opcoes = new List<string>();

    public void AdicionarOpcao(string opcao)
    {
        opcoes.Add(opcao);
    }

    public void Mostrar()
    {
        Console.WriteLine("\n=== MENU ===");
        for (int i = 0; i < opcoes.Count; i++)
        {
            Console.WriteLine(i + " - " + opcoes[i]);
        }
    }

    public int Escolher()
    {
        return int.Parse(Console.ReadLine());
    }
}